libpod/runtime_pod_linux.go

```diff
@@ -13,8 +13,14 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+// podIsSharing checks if the pod has any kernel namespaces set as shared. A pause container will not be
+// created if no kernel namespaces are shared.
+func podIsSharing(p *Pod) bool {
+	return p.UsePodPID() || p.UsePodIPC() || p.UsePodNet() || p.UsePodMNT() || p.UsePodUser() || p.UsePodUTS()
+}
+
 // NewPod makes a new, empty pod
-func (r *Runtime) NewPod(options ...PodCreateOption) (*Pod, error) {
+func (r *Runtime) NewPod(ctx context.Context, usePause bool, options ...PodCreateOption) (*Pod, error) {
```

Comment:
I wouldn't make usePause a separate bool - have a PauseContainer struct in the pod config with a MakePauseContainer bool in it, and set it with a PodCreateOption.